tests/src/JIT/Directed/pinvoke/user32menu.cpp

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// This file fakes a few Windows APIs to enable the tail.il test on all platforms.
+
+#include <cstring>
+#include <string>
+#include <vector>
+
+#if defined(_MSC_VER)
+#define EXPORT_API extern "C" __declspec(dllexport)
+#else
+#define EXPORT_API extern "C" __attribute__((visibility("default")))
+
+#ifdef BIT64
+#define __int64     long
+#else // BIT64
+#define __int64     long long
+#endif // BIT64
+
+#define __int32     int
+#define __int16     short int
+#define __int8      char        // assumes char is signed
+
+#endif 
+
+#include <cstddef>
+
+typedef std::vector<std::string> MENU;
+typedef MENU * HMENU;
+
+EXPORT_API
+HMENU
+CreatePopupMenu()
+{
+    return new MENU();
+}
+
+EXPORT_API
+unsigned __int32
+DestroyMenu(
+    HMENU hMenu
+    )
+{
+    delete hMenu;
+
+    return 1;
+}
+
+EXPORT_API
+unsigned __int32
+AppendMenuA(
+    HMENU hMenu,
+    unsigned __int32 uFlags,
+    unsigned __int32 uID,
+    const char * item
+    )
+{
+    if (uFlags != 0)
+    {
+        throw "AppendMenu: only MF_STRING (0) supported for uFlags";
```

Comment:
Throwing like this will result in a very bad failure on non-Windows. I doubt the string will be observed in the logs. That may be okay in this case, but just FYI.

Reply:
Can you clarify "very bad failure?"  I agree this is low effort error handling, but I really just need to indicate catastrophic failure.  The only "very bad" failure modes IMO would be if the throws are ignored or cause the process to hang.  Can you point out a better example I should follow?  I'm happy to do something else--just not sure what is preferred.

Reply:
>Can you clarify "very bad failure?"

The process will simply explode with no indicate this is the cause since as soon as the transition back to the CLR occurs the process stops with an ugly segv or something else that isn't obvious. I think I should have clarified my point a bit better. If you want this message to be in the log so it is obvious why it failed, I would instead write out the message using `printf` and then throw, that was my main point - this message will never be observed in the log.

Reply:
Yeah, I don't care whether the message is in the log.  It's really just taking the place of a comment in the code so I type less characters.